mlflow/pyfunc/scoring_server/client.py

```diff
@@ -0,0 +1,143 @@
+import requests
+import time
+from mlflow.pyfunc import scoring_server
+import subprocess
+import os
+import sys
+import signal
+import json
+
+
+class ScoringServerClient:
+
+    def __init__(self, host, port):
+        self.url_prefix = f"http://{host}:{port}"
+
+    def ping(self):
+        ping_status = requests.get(url=self.url_prefix + "/ping")
+        if ping_status.status_code != 200:
+            raise Exception(f"ping failed (error code {ping_status.status_code})")
+
+    def wait_server_ready(self, timeout=30):
+        begin_time = time.time()
+        while True:
+            time.sleep(0.3)
+            try:
+                self.ping()
+                return
+            except Exception:
+                pass
+            if time.time() - begin_time > timeout:
+                break
+        raise RuntimeError('Wait scoring server ready timeout.')
+
+    def get_module_version(self, module_name):
+        """
+        Get module version on the scoring server worker python environment.
+        This method is for testing purpose.
+        """
+        status = requests.get(url=self.url_prefix + f"/version/{module_name}")
+        if status.status_code != 200:
+            raise Exception(f"get_module_version failed (error code {status.status_code})")
+        return status.text.strip()
+
+    def invoke(self, data, pandas_orient="records"):
+        import pandas as pd
+
+        content_type_list = []
+        if type(data) == pd.DataFrame:
+            content_type_list.append(scoring_server.CONTENT_TYPE_JSON)
+            if pandas_orient == "records":
+                content_type_list.append(scoring_server.CONTENT_TYPE_FORMAT_RECORDS_ORIENTED)
+            elif pandas_orient == "split":
+                content_type_list.append(scoring_server.CONTENT_TYPE_FORMAT_SPLIT_ORIENTED)
+            else:
+                raise Exception(
+                    "Unexpected pandas_orient for Pandas dataframe input %s" % pandas_orient
+                )
+        else:
+            raise RuntimeError("Unsupported data type.")
+
+        post_data = json.dumps(scoring_server._get_jsonable_obj(data, pandas_orient=pandas_orient))
+
+        response = requests.post(
+            url=self.url_prefix + "/invocations",
+            data=post_data,
```

Comment:
```suggestion
        post_data = scoring_server._get_jsonable_obj(data, pandas_orient=pandas_orient)

        response = requests.post(
            url=self.url_prefix + "/invocations",
            data=post_data,
```

Can we directly pass a dict to `requests.post`

Reply:
No. In my test it failed.

Reply:
Ah, I thought `post_data` is a dictionary but it's not, right?

Reply:
This seems related to the content type.